lib/utils/job-utils/racadm-tool.js

```
@@ -0,0 +1,221 @@
+     * @param {string} password
+     * @param {string} jobId -  job id created by iDRAC
+     */
+    RacadmTool.prototype.getJobStatus = function(host, user, password, jobId) {
+        return this.runCommand(host, user, password, "jobqueue view -i " + jobId)
+            .then(function(consoleOutput){
+                return parser.getJobStatus(consoleOutput);
+            });
+    };
+
+    /**
+     * Returns a promise with the software inventory
+     *
+     * @param {string} host
+     * @param {string} user
+     * @param {string} password
+     * @param {string} jobId
+     * @param {number} retryCount - retry count
+     * @param {number} delay - delay time for retry
+     */
+    RacadmTool.prototype.waitJobDone = function(host, user, password, jobId, retryCount, delay) {
+        var self = this;
+        return self.getJobStatus(host, user, password, jobId)
+            .then(function(jobStatus){
+                if (jobStatus.status === 'Completed'){
+                    return jobStatus;
+                } else if (jobStatus.status === 'Failed'){
+                    throw {error: 'Job Failed during process',
+                        jobStatus: jobStatus};
+                } else if (jobStatus.status === 'Running'){
+                    if (retryCount < self.retries) {
+                        return Promise.delay(delay)
+                            .then(function () {
+                                retryCount += 1;
+                                delay = 2 * delay;
+                                return self.waitJobDone(host, user, password,
+                                    jobId, retryCount, delay);
+                            });
+                    } else {
+                        throw {error: 'Job Timeout',
+                            jobStatus: jobStatus};
+                    }
+                } else{
+                    throw {error: 'Job status is incorrect',
+                        jobStatus: jobStatus};
```

should throw an Error instance